test/channel.js

```diff
@@ -225,6 +225,39 @@ test('event channel', assert => {
   assert.end()
 });
 
+test('unsubscribe event channel', assert => {
+  let unsubscribed = false;
+  let chan = eventChannel(() => () => {
+    unsubscribed = true;
+  });
+  chan.close();
+  assert.ok(unsubscribed, 'eventChannel should call unsubscribe when channel is closed')
+
+  unsubscribed = false;
+  chan = eventChannel((emitter) => {
+    emitter(END);
+    return () => {
+      unsubscribed = true;
+    }
+  });
+  assert.ok(unsubscribed, 'eventChannel should call unsubscribe when END event is emitted synchronously');
+
+  unsubscribed = false;
+  chan = eventChannel((emitter) => {
+    setTimeout(() => emitter(END), 0);
+    return () => {
+      unsubscribed = true;
+    }
+  });
+  chan.take(input => {
+    // console.log(input, unsubscribed);
+    assert.equal(input, END, 'should emit END event');
+    // assert.ok(unsubscribed, 'eventChannel should call unsubscribe when END event is emitted asynchronously');
```

Comment:
i can't get this test to pass without using setTimeout, any thoughts @Andarist ?